packages/loader/container-loader/src/deltaManager.ts

```diff
@@ -1012,13 +1023,17 @@ export class DeltaManager
                 if (result.done) {
                     break;
                 }
-                callback(result.value);
+                try {
+                    opsFromFetch = true;
+                    callback(result.value);
+                } finally {
+                    opsFromFetch = false;
+                }
             }
         } finally {
+            assert(!opsFromFetch, "logic error");
```

Comment:
Is it worth asserting on this? Not sure if without code changes this can ever be `true` at this point. If this is the intention then please ignore :)

Reply:
Right, the purpose of asserts is to validate constraints / invariants. In this case it validates invariant that we correctly use RAII pattern around callback.